src/Components/Blazor/Blazor/src/Http/WebAssemblyHttpMessageHandler.cs

```diff
@@ -168,14 +188,21 @@ public HttpResponseMessage ToResponseMessage(HttpContent content)
                 var contentHeaders = result.Content?.Headers;
                 foreach (var pair in Headers)
                 {
-                    if (!headers.TryAddWithoutValidation(pair[0], pair[1]))
+                    if (!headers.TryAddWithoutValidation(pair.Name, pair.Value))
                     {
-                        contentHeaders?.TryAddWithoutValidation(pair[0], pair[1]);
+                        contentHeaders?.TryAddWithoutValidation(pair.Name, pair.Value);
                     }
                 }
 
                 return result;
             }
         }
+
+        private class Header
```

Comment:
readonly struct?

Reply:
System.Text did not support value types when I tried this.

Reply:
Now it supports value types as well as reference types.

Reply:
Yep, its fixed: https://github.com/dotnet/corefx/issues/37434#issuecomment-494635210

Reply:
Unless I'm mistaken, this using a `readonly struct` would not work since the serializer does not support deserializing in to ctor arguments. Plus it looks like you end up boxing the struct on deserialization, so it seems less useful. Might be more useful to switch to KVP if the serializer natively supports it: https://github.com/dotnet/corefx/issues/36643?

Reply:
ok well let's leave it as-is then.